Resources/doc/4-integrating_fosub.md

```diff
@@ -0,0 +1,197 @@
+     * @return User
+     */
+    public function setFacebookAccessToken($facebookAccessToken)
+    {
+        $this->facebook_access_token = $facebookAccessToken;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFacebookAccessToken()
+    {
+        return $this->facebook_access_token;
+    }
+```
+After adding extra properties to User entity, you need to extend base FOSUBUserProvider (if you want to add more advanced behavior, than provided from the box)
+
+
+### 2) Extend default FOSUBUserProvider
+
+The bundle provide bridge class for connect FOSUserBundle User class and HWIOAuth out of the box.
+You should extend it if you want to add more advanced behavior.
+
+In `YourBundle\Security\Core\User` create class, lets call it `MyFOSUBUserProvider`:
+
+```php
+
+namespace MyBundle\Security\Core\User;
+
+use HWI\Bundle\OAuthBundle\OAuth\Response\UserResponseInterface;
+use HWI\Bundle\OAuthBundle\Security\Core\User\FOSUBUserProvider as BaseFOSUBProvider;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class FOSUBUserProvider extends BaseFOSUBProvider
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(UserManagerInterface $userManager, array $properties)
+    {
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function connect(UserInterface $user, UserResponseInterface $response)
+    {
+        $property = $this->getProperty($response);
+        $username = $response->getUsername();
+        $service = $response->getResourceOwner()->getName(); //get name of resource owner
+
+        //we "disconnect" previously connected users
+        $previousUser = $this->userManager->findUserBy([$property => $username])
+        if (null !== $previousUser) {
+            // set current user id and token to null for disconect
+            ...
+            $this->userManager->updateUser($previousUser);
+        }
+        //we connect current user, set current user id and token
+        ...
+        $this->userManager->updateUser($user);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function loadUserByOAuthUserResponse(UserResponseInterface $response)
+    {
+        $username = $response->getRealName();
+        $userEmail = $response->getEmail();
+        $user = $this->userManager->findUserByEmail($userEmail);
+        if (null === $user) {
+            // if null just create new user and set it properties
+
+            return $user
+        }
+        // else update access token of existing user
+        $serviceName = $response->getResourceOwner()->getName();
+        $setter = 'set' . ucfirst($serviceName) . 'AccessToken';
+        $user->$setter($response->getAccessToken());//update access token
+
+        return $user;
+    }
+}
+
+```
+
+### 3) Configure user provider as service
+
+To `MyBundle\Resources\config\services.xml`, add following code for declare your custom provider as service
```

Comment:
This will not work if someone in that bundle don't add extension class, IMO for documentation it can be suggested to add in main `config.yml`...